patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -207,6 +221,33 @@
 +    protected ThreadLocal<EntityPlayer> harvesters = new ThreadLocal();
 +    private ThreadLocal<IBlockState> silk_check_state = new ThreadLocal();
 +    protected static java.util.Random RANDOM = new java.util.Random(); // Useful for random things without a seed.
++
++    /**
++     * Returns whether or not this block is of a type that needs random ticking.
++     * Called for ref-counting purposes by ExtendedBlockStorage in order to
++     * broadly cull a chunk from the random chunk update list for efficiency's sake.
++     *
++     * @param state The current block state
++     * @return True if the block should tick randomly
++     */
++    public boolean shouldTickRandomly(IBlockState state)
++    {
++        return func_149653_t();
++    }
++
++    /**
++     * Gets how many world ticks before ticking
++     *
++     * @param world The current world
++     * @param pos Block position in
++     * @param state The current block state
++     * @return The tick rate
++     */
++    public int getTickRate(World world, BlockPos pos, IBlockState state)
```

Comment:
Like above, a lot of the docs in here don't really add any useful information that you don't get just by reading the signature or the parameter names/function name.

I'd change it to
```java
/**
 * @return How many world ticks should occur before ticking
 */
```

Although it isn't really that clear still. How and when would someone want to use this? What do you mean by `ticking`?